python/ray/monitor.py

```diff
@@ -468,6 +475,77 @@ def driver_removed_handler(self, unused_channel, data):
 
         self._clean_up_entries_for_driver(driver_id)
 
+    def _xray_clean_up_entries_for_driver(self, driver_id):
+        """Remove this driver's object/task entries from redis.
+
+        Removes control-state entries of all tasks and task return
+        objects belonging to the driver.
+        """
+        redis = self.state.redis_client
+
+        task_table_infos = {}  # task id -> TaskInfo
+        for key in redis.scan_iter(match=XRAY_TASK_TABLE_PREFIX + b"*"):
+            entry = redis.get(key)
+            task = Task.GetRootAsTask(entry, 0)
```

Comment:
Can you actually parse it this way? Don't you need to parse it as a GcsTableEntry first?